### DISCUSSION TITLE
Multiselect json field in WhereIn clause

### DISCUSSION THREAD

**Discussion by jasha1974** [filamentphp/filament#5686] [Help]
Hi guys, need help with query. I have multiselect json field, on model cast to array and livewire checkboxes wire:model as array. How to put to whereIN clause on query ?

thanx a lot

**Comment by danharrin** [comment]
Pass the array as the second parameter of `whereIn()`

**Reply by jasha1974** [reply]
it doesn't work, where clause `filters` in ('4', '5'), but json data in db ["4", "5"]

**Reply by danharrin** [reply]
Your question is vague with no code example. If you'd like a more detailed answer, please ask a more detailed question.

**Reply by jasha1974** [reply]
i'v got table filters for examples for men (id 1), for woman (id 2), color black (id 3), color white (id 4). Product has multiselect json field belongs to id 1, 3. Livewire checkboxes filter returns array 1, 4. how to compare product and checkboxes in query? is it clear?

**Reply by jasha1974** [reply]
code 


   public function updatedFilters()
    {
        

        if (!is_array($this->filters)) return;

        $this->filters = array_filter($this->filters, function ($filter) {
            return $filter != false;
        });


    }

    public function render()
    {

        $products = Product::Search($this->search)

                    ->when($this->filters, function ($q, $filters) {

	               $q->WhereIN('filters', array_values($filters));

                    })

                    ->paginate($this->pagesize);



        return view('livewire.shop-component', [

            'products' => $products,

        ])->layout('layouts.app');
    }

**Reply by danharrin** [reply]
What is `dd(array_values($filters))`

**Reply by jasha1974** [reply]
from Log::info. 
local.INFO: array (
  0 => '4',
  1 => '5',
)

**Reply by danharrin** [reply]
And what format is data inside the `filters` column in your database

**Reply by jasha1974** [reply]
["9", "5"]

**Reply by danharrin** [reply]
Try `whereJsonContains()` instead of `whereIn()`

**Reply by jasha1974** [reply]
whereJsonContains or orWhereJsonContains is not what i need. first case if i select men and woman (id 1 and 2) return null and for second case need filter all for man and color black only. it's understandable ?

                        $q->orwhereJsonContains('filters', array_values($filters));

                        for ($i = 0; $i < count($filters); $i++) {
                            $q->orWhereJsonContains('filters', array_values($filters)[$i]);
                        }

**Reply by danharrin** [reply]
I'm sorry but this is not a Filament issue, its something you will need to debug yourself or ask in a Laravel forum

**Reply by jasha1974** [reply]
ok, dan, but thanx a lot
